src/Simplify.elm

```diff
@@ -3605,6 +3719,228 @@ listIndexedMapChecks { lookupTable, fnRange, firstArg } =
                     []
 
 
+listFoldlChecks : CheckInfo -> List (Error {})
+listFoldlChecks checkInfo =
+    listFoldAnyDirectionChecks "foldl" checkInfo
+
+
+listFoldrChecks : CheckInfo -> List (Error {})
+listFoldrChecks checkInfo =
+    listFoldAnyDirectionChecks "foldr" checkInfo
+
+
+listFoldAnyDirectionChecks : String -> CheckInfo -> List (Error {})
+listFoldAnyDirectionChecks foldOperationName checkInfo =
+    -- TODO discuss adding
+    -- List.fold (&|) initial list --> initial &| (List.all identity list)
+    --         and piping versions
+    -- List.fold (&|) initial --> (List.all identity >> (&|) initial)
+    --     Are those simplifications? arguable, I'm leaning no
+    --     At the very least, they are more performant (stoppable).
+    -- List.fold f x [ a ] --> f (a) x
+    --         @jfmengels' comment: potentially quite ugly
+    --         I'd say it looks fine
+    case ( checkInfo.secondArg, checkInfo.thirdArg ) of
+        ( Just (Node initialArgumentRange _), Just (Node _ (Expression.ListExpr [])) ) ->
+            [ Rule.errorWithFix
+                { message = "The call to List." ++ foldOperationName ++ " will result in the initial accumulator"
+                , details = [ "You can replace this call by the initial accumulator." ]
+                }
+                checkInfo.fnRange
+                [ Fix.removeRange
+                    { start = initialArgumentRange.end, end = checkInfo.parentRange.end }
+                , Fix.removeRange
+                    { start = checkInfo.parentRange.start, end = initialArgumentRange.start }
```

Comment:
Can you change the order of these from earlier in the file to later in the file? This was confusing to me :sweat_smile: 

This will be sorted that way anyway, so no need to do it the opposite way for performance.

Reply:
🤯 It's those little things that elm-review silently does